data_structures/stack_using_queue.cpp

```diff
@@ -0,0 +1,122 @@
+/**
+ * @file
+ * @brief Implement a last in first out (LIFO) stack using queues. The implemented stack should support all the functions of a normal queue (push, top, pop, and empty).
```

Comment:
Provide a Wikipedia link in Markdown format (if available/possible).
If there's no Wikipedia link, add another web source for algorithm explanation. 🙂